aws-s3/region.mli

```diff
@@ -1,3 +1,9 @@
+type vendor = {
```

Comment:
I must admit that don't like to expose this datastructure in the API. Maybe a function to create the region would shield would be better:
```
val make_vendor: region_name:string -> host:string -> port:int -> t
```
And possibly also adding 
```
val minio: host:string -> port:int -> t
```